bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -289,34 +336,77 @@ private void logTraceWithPattern(int responseStatus, String content) {
     }
 
     private boolean areWeLoggedIn() throws ExecutionException, InterruptedException, TimeoutException {
-        logger.debug("Checking if we are logged in");
-        String url = STATUS;
+        logger.debug("Checking if session is valid");
+
+        int statusCode = checkSessionStatus();
+        if (statusCode == HttpStatus.OK_200) {
+            logger.debug("Session is valid");
+            return true;
+        } else if (statusCode == HttpStatus.UNAUTHORIZED_401 && !vsRefresh.isEmpty()) {
+            logger.debug("Session expired, attempting token refresh");
+            if (refreshTokens()) {
+                statusCode = checkSessionStatus();
+                return statusCode == HttpStatus.OK_200;
+            } else {
+                logger.debug("Token refresh failed, will need to perform full login");
+                vsRefresh = "";
+                return false;
+            }
+        } else {
+            logger.debug("Session invalid (status code: {}), need to login", statusCode);
+            return false;
+        }
+    }
 
-        ContentResponse response = httpClient.newRequest(url).method(HttpMethod.GET).send();
-        String content = response.getContentAsString();
-        logTraceWithPattern(response.getStatus(), content);
+    private int checkSessionStatus() throws ExecutionException, InterruptedException, TimeoutException {
+        String url = START_GRAPHQL;
+        String queryQLAccountInstallations = "[{\"operationName\":\"AccountInstallations\",\"variables\":{\"email\":\""
+                + userName
+                + "\"},\"query\":\"query AccountInstallations($email: String!) {\\n  account(email: $email) {\\n    owainstallations {\\n      giid\\n      alias\\n      type\\n      subsidiary\\n      dealerId\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n\"}]";
```

Comment:
Consider a [Java Text Clock](https://www.baeldung.com/java-text-blocks) for more readable formatting.

Reply:
Fixed!